src/main/java/com/couchbase/lite/replicator/BlobRequestBody.java

```diff
@@ -42,6 +44,36 @@ public MediaType contentType() {
                 return contentType;
             }
 
+
+            /**
+             * OkHttp use chunked transfer if content-length is unknown. And CouchDB 1.6.1 can not
+             * handle chunked transer with multipart doc, then CouchDB disconnect .
+             * To send non-chunked transfer with multipart document by OkHttp, the client needs
+             * to specify content-length.
+             * Sync Gateway can accept chunked transfer
+             */
+            @Override
+            public long contentLength() throws IOException {
+                // Sync Gateway
+                if (syncGateway) {
+                    return super.contentLength();
+                }
```

Comment:
Is chunked transfer the preferred method by SyncGateway? Can we have only one logic?